src/alire/alire-solutions.adb

```diff
@@ -287,6 +289,53 @@ package body Alire.Solutions is
       end case;
    end Is_Better;
 
+   -------------
+   -- Linking --
+   -------------
+
+   function Linking (This  : Solution;
+                     Crate : Crate_Name;
+                     Link  : Externals.Softlinks.External)
+                     return Solution
+   is
+      use Alire.OS_Lib.Operators;
+
+      ----------
+      -- Join --
+      ----------
+
+      function Join (Parent, Child : Any_Path) return Any_Path
+      is (if Check_Absolute_Path (Child)
+          then Child
+          else Parent / Child);
+
+      Linked_Root : constant Roots.Optional.Root :=
+                      Roots.Optional.Detect_Root (Link.Path);
+   begin
+
+      --  Recursively find any other links
+
+      return Result : Solution := (Solved       => True,
+                                   Dependencies =>
+                                     This.Dependencies.Including
+                                       (This.State (Crate).Linking (Link)))
+      do
+         if Linked_Root.Is_Valid and then Linked_Root.Value.Has_Lockfile then
+            for Dep of Linked_Root.Value.Solution.Links loop
+               Result := Result
```

Comment:
This statement is not very readable in my opinion.

Maybe the result of `Externals.Softlinks.New_Softlink` could be in constant.
same for `Linked_Root.Value.Solution.State`.

Reply:
I have refactored things a bit for clarity.